File: 01.两数之和/Go语言/towNumsSum.go
package main

import "fmt"

/**
方法一：暴力法
*/
func twoSum1(nums []int, target int) []int {
	for i := 0; i < len(nums)-1; i++ {
		for j := 1; j < len(nums); j++ {
			if nums[i]+nums[j] == target {
				return []int{i, j}
			}
		}
	}
	return []int{}
}

/**
方法二：哈希表法
*/
func twoSum2(nums []int, target int) []int {
	map1 := make(map[int]int)
	for i, v := range nums {
		//如果存在则返回
		if _, ok := map1[target-v]; ok {
			return []int{map1[target-v], i}
		}
		//不存在则存入，key为nums的值，value为nums的索引
		map1[nums[i]] = i
	}
	return []int{}
}

func main() {
	nums := []int{2, 7, 11, 15, 17}
	target := 24
	// result1 := twoSum1(nums, target)
	// fmt.Println("result:", result1)
	result2 := twoSum2(nums, target)
	fmt.Println("result2:", result2)
}

File: 01.两数之和/Java语言/towNumsSum.java
import java.util.HashMap;

class towNumsSum {
  /**
   * 方法一：暴力法
   */
  public static int[] twoSum1(int[] nums, int target) {
    for (int i = 0; i < nums.length - 1; i++) {
      for (int j = i + 1; j < nums.length; j++) {
        if (nums[i] + nums[j] == target) {
          return new int[] { i, j };
        }
      }
    }
    return new int[] {};
  }

  /**
   * 方法二： 哈希表法
   */
  public static int[] twoSum2(int[] nums, int target) {
    HashMap<Integer, Integer> map = new HashMap<>();
    for (int i = 0; i < nums.length; i++) {
      if (map.containsKey(target - nums[i])) {
        return new int[] { map.get(target - nums[i]), i };
      }
      map.put(nums[i], i);
    }
    return new int[] {};
  }

  public static void main(String[] args) {
    int[] nums = { 2, 7, 11, 15 };
    int target = 18;
    int[] result1 = twoSum1(nums, target);
    System.out.println("[" + result1[0] + "," + result1[1] + "]");
    int[] result2 = twoSum2(nums, target);
    System.out.println("[" + result2[0] + "," + result2[1] + "]");
  }
}
